app/taco.ts

```diff
@@ -0,0 +1,97 @@
+import {conditions, decrypt, domains, encrypt, ThresholdMessageKit} from '@nucypher/taco';
+import { EIP4361AuthProvider, USER_ADDRESS_PARAM_DEFAULT } from '@nucypher/taco-auth';
+import {ethers} from "ethers";
+
+let authManager = {
+    provider: null,
+    authProvider: null,
+    currentAccount: null,
+
+    async initialize() {
+      if (!this.provider) {
+        this.provider = new ethers.providers.Web3Provider(window.ethereum);
```

Comment:
Just to confirm, so we never need to re-instantiate a provider, even when the account changes? (Pardon my ignorance of ethers)

Reply:
That's certainly my understanding of how ethers works